SPECS/td-agent/td-agent.spec

```diff
@@ -0,0 +1,217 @@
+    gzip ${man}
+done
+
+cd -
+mkdir -p %{buildroot}%{_localstatedir}/run/td-agent
+mkdir -p %{buildroot}%{_localstatedir}/log/td-agent
+mkdir -p %{buildroot}%{_localstatedir}/log/td-agent/buffer
+mkdir -p %{buildroot}%{_sysconfdir}/td-agent/plugin
+mkdir -p %{buildroot}/tmp/td-agent
+
+%pre
+sudo ln -S %{_sbindir}/useradd %{_sbindir}/adduser
+if ! getent group td-agent >/dev/null; then
+    %{_sbindir}/groupadd -r td-agent
+fi
+if ! getent passwd td-agent >/dev/null; then
+    %{_sbindir}/adduser -r -g td-agent -d %{_localstatedir}/lib/td-agent -s /sbin/nologin -c 'td-agent' td-agent
+fi
+
+%preun
+if [ $1 -eq 0 ]; then
+  %{control_service stop td-agent}
+  %{disable_service disable td-agent}
+fi
+
+%post
+if [ $1 -eq 1 ]; then
+  %{enable_service enable td-agent}
+fi
+if [ $1 -eq 2 ]; then
+  %{control_service condrestart td-agent}
+fi
+if [ -d "%{_sysconfdir}/prelink.conf.d/" ]; then
+  echo "prelink detected. Installing %{_sysconfdir}/prelink.conf.d/td-agent-ruby.conf ..."
+  if [ ! -f %{_sysconfdir}/prelink.conf.d/td-agent-ruby.conf ]; then
+      cp -f /opt/td-agent/share/td-agent-ruby.conf %{_sysconfdir}/prelink.conf.d/td-agent-ruby.conf
+  else
+    if [ $(grep '\-b /opt/td-agent/embedded/bin/ruby' -c %{_sysconfdir}/prelink.conf.d/td-agent-ruby.conf) -eq 1 ]; then
+      echo "old-prelink detected, Update %{_sysconfdir}/prelink.conf ..."
+      sed -i"" %{_sysconfdir}/prelink.conf.d/td-agent-ruby.conf -e "s,/embedded/,/,"
+    fi
+  fi
+elif [ -f "%{_sysconfdir}/prelink.conf" ]; then
+  if [ $(grep '\-b /opt/td-agent/embedded/bin/ruby' -c %{_sysconfdir}/prelink.conf) -eq 1 ]; then
+    echo "old-prelink detected, but %{_sysconfdir}/prelink.conf.d/ does't exist. Update %{_sysconfdir}/prelink.conf ..."
+    sed -i"" %{_sysconfdir}/prelink.conf -e "s,/embedded/,/,"
+  elif [ $(grep '\-b /opt/td-agent/bin/ruby' -c %{_sysconfdir}/prelink.conf) -eq 0 ]; then
+    echo "prelink detected, but %{_sysconfdir}/prelink.conf.d/ does't exist. Adding %{_sysconfdir}/prelink.conf ..."
+    echo "-b /opt/td-agent/bin/ruby" >> %{_sysconfdir}/prelink.conf
+  fi
+fi
+sudo systemctl start td-agent
+
+%files
+%doc README.md
+%license LICENSE
+%defattr(-,root,root,-)
+/opt/*
+/opt/td-agent/share/td-agent.conf.tmpl
+%{_prefix}/%{_unitdir}/td-agent.service
+%{_tmpfilesdir}/td-agent.conf
+%{_bindir}/td
+%{_sbindir}/td-agent
+%{_sbindir}/td-agent-gem
+%config(noreplace) %{_sysconfdir}/td-agent/td-agent.conf
+%config(noreplace) %{_sysconfdir}/logrotate.d/td-agent
+%attr(0755,td-agent,td-agent) %dir %{_localstatedir}/log/td-agent
+%attr(0755,td-agent,td-agent) %dir %{_localstatedir}/log/td-agent/buffer
+%attr(0755,td-agent,td-agent) %dir %{_sysconfdir}/td-agent
+%attr(0755,td-agent,td-agent) %dir %{_sysconfdir}/td-agent/plugin
+%attr(0755,td-agent,td-agent) %dir /tmp/td-agent
+
+%changelog
+* Mon Jan 04 2021 Henry Li <lihl@microsoft.com> - 4.0.1-1
+- Initial CBL-Mariner import from Treasure Data (license: Apache2).
```

Comment:
Please also check if the `License` tag aligns with the license of the project.

Reply:
Verified